### ISSUE TITLE
`mse_loss` issue if one of the inputs `requires_grad`

### ISSUE THREAD

**Issue by cheind** [pytorch/pytorch#11149]
When one of the inputs to `torch.nn.functional.mse_loss` requires gradients the returned loss changes from elementwise mean to sum.

```
In [43]: import torch
    ...: from torch.nn.functional import mse_loss
    ...:
    ...: x = torch.rand(1,2,3)
    ...: y = torch.rand(1,2,3, requires_grad=True)
    ...:
    ...: print(mse_loss(x,y, reduction='elementwise_mean'))
    ...: print(((x-y)**2).mean())
    ...:
    ...:
tensor(0.7133, grad_fn=<SumBackward0>)
tensor(0.1189, grad_fn=<MeanBackward1>)
```
## System Info

PyTorch version: 0.4.1
Is debug build: No
CUDA used to build PyTorch: 9.0

OS: Microsoft Windows 10 Home
GCC version: Could not collect
CMake version: version 3.7.1

Python version: 3.6
Is CUDA available: Yes
CUDA runtime version: 9.0.176
GPU models and configuration: GPU 0: GeForce GTX 1070
Nvidia driver version: 388.13
cuDNN version: Could not collect
